src/ipc/handler.c

```diff
@@ -70,8 +70,16 @@
 #define iGS(x) ((x >> SOF_GLB_TYPE_SHIFT) & 0xf)
 #define iCS(x) ((x >> SOF_CMD_TYPE_SHIFT) & 0xfff)
 
-#define IPC_INVALID_SIZE(ipc) \
-	(sizeof(*(ipc)) != ipc->hdr.size)
+/*
+ * ABI version compatibility rules :-
+ *
+ * 1) Host drivers will support the current and some older MAJOR versions of
```

Comment:
And all MAJOR versions of the IPC ABI (up to a certain age to be determined by market information)